src/modeler/AutoRest.Swagger/Validation/BodyPropertiesNamesCamelCase.cs

```diff
@@ -60,7 +60,9 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<
                             {
                                 if (!ValidationUtilities.IsODataProperty(prop.Key) && !ValidationUtilities.IsNameCamelCase(prop.Key))
                                 {
-                                    yield return new ValidationMessage(new FileObjectPath(context.File, context.Path), this, prop.Key, ValidationUtilities.GetCamelCasedSuggestion(prop.Key));
+                                    yield return new ValidationMessage(new FileObjectPath(context.File, 
+                                        context.Path.AppendProperty(operation).AppendProperty("parameters").AppendIndex(path[operation].Parameters.IndexOf(param)).AppendProperty("schema").AppendProperty("properties").AppendProperty(prop.Key)), 
```

Comment:
I'd switch the `foreach` loop to a `for` loop rather than using `IndexOf`! That's a cheap optimization and less error prone.